lib/utils/fs-utils.js

```diff
@@ -46,12 +48,56 @@ module.exports = (function(){
             }, function() {
                 return false;
             });
+    }   
+
+    function updateFileModifiedTime(fullPath, fileModifiedTime) {
+
+       try {
+            fs.utimes(fullPath, fileModifiedTime, fileModifiedTime, function (err) {
+                if (err) {
+                    logger.error('error: %s, failed to update modified time of %s to %s', errorUtils.error2string(err), fullPath, fileModifiedTime.toDateString());
+                }
+                else {
+                    logger.debug('successfully updated modified time of %s to %s', fullPath, fileModifiedTime.toDateString());
+                }
+            });
+
+        } catch (e) {
+            logger.error('exception, failed to set modified time of %s to %s. error: %s', fullPath, fileModifiedTime.toDateString(), errorUtils.error2string(e));
+        }
+
+    }
+
+    function checkIfFileExpired(fullPath, fileExpireAge) {
+
+        return Q.Promise(function(resolve, reject) {
```

Comment:
i recommend not todo promise in this but do regular callback due to performance reasons